### ISSUE TITLE
Events no longer working; RegEx mismatch.

### ISSUE THREAD

**Issue by fjeddy** [garrettjoecox/scriptserver-event#4]
Seems like the logging layout has changed and therefor the regex no longer fit. Unless I've misunderstood something and are supposed to do something to log it differently / correctly?

Using Spigot.

**Comment by fjeddy** [comment]
Hey you,

Running the latest version, 1.12.2 with Spigot. All of the events seems to be broken. Tho, I didn't test the shutdown / started ones.

**Comment by fjeddy** [comment]
I really should test different versions before I start engaging the world.

It works perfectly fine with Vanilla Minecraft, tho, it fails with the most popular builds, Bukkit and Spigot (Which I believe are even more used then the vanilla). Tho, scriptserver-json is spitting out the following error when used with essentials

```
fs.js:132
    throw new ERR_INVALID_CALLBACK();
    ^

TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
    at makeCallback (fs.js:132:11)
    at Object.fs.mkdir (fs.js:822:14)
    at fs.stat (/Users/freddy/Sites/*/*.public/node_modules/scriptserver-json/index.js:13:17)
    at FSReqWrap.oncomplete (fs.js:149:21)
```

Might be related to me running things with await instead of callbacks?

Really sorry for wasting your time.

**Comment by garrettjoecox** [comment]
No worries, not waisting my time! Looking into using spigot right now.

**Comment by garrettjoecox** [comment]
Alright, there seems to be two issues with spigot:
1) The logging for "login" and "logout" events differ from vanilla to spigot, breaking the regex in this module.
2) The RCON log differs from vanilla to spigot, breaking the RCON connection in the main module. (preventing use of `.send()`)

I would recommend using a vanilla version of minecraft with ScriptServer until I am able to sort these out.

**Comment by fjeddy** [comment]
Wait, what? Annoying that they had to make that different, but what exactly does the logging have to do with executing commands? (Got me really curious now) :P

**Comment by garrettjoecox** [comment]
In the core module, I have to wait for the server's RCON listener to start before I am able to connect with it (Once it's started it logs out "RCON running on ___")

**Comment by fjeddy** [comment]
Ahh, I see. Thanks for the effort, appreciate it.

Unfortunately, vanilla is not an option for me due to the lack of plugins and API's that Bukkit and Spigot is known for. This is a rather larger server as well, so the performance difference between Spigot and Vanilla is very noticeable, where Vanilla is requiring a lot more.

Will see if I can figure it out and help with some PR's :)

**Comment by garrettjoecox** [comment]
Ya for sure, definitely need this to support 3rd party server implementations, just never got around to testing it.

**Comment by garrettjoecox** [comment]
Proposed Solutions:
- Support both logging statements (likely will break with other server implementations)
- Support custom REGEX, allowing user to fix the issue (requires user to know what they are doing..)
- Support specifying server type (vanilla/spigot/bukkit) in the config and changing out the regex based on it.

**Comment by fjeddy** [comment]
Option number three is what I had in mind, and what I would do if I where to do this.

Spigot is a drop-in replacement for Bukkit, I have to double-check and confirm, but I'm 98% sure that they are identical on this area, so just one set of Regex is needed there.

**Comment by garrettjoecox** [comment]
Solved with commit 5003d938bf1d2bb6219937729438f0b5efb3d784
